# **2022 NFL Combine - Project 3**
## Inference for the Population Proportion

# **Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that add extra features to the basic setup. The name of the modules is after the `import` statement, and the purpose is in a non-code comment after thew hashtag (#).



In [14]:
import pandas as pd                 #Data analysis
import numpy as np                  #Calculations
from IPython.display import Image   #Display images
from scipy.stats import norm        #Confidence Interval

In [15]:
# Assigns the URL of the image to display to the name 'image_url'.
image_url = 'https://as2.ftcdn.net/v2/jpg/04/31/74/37/1000_F_431743763_in9BVVzCI36X304StR89pnxyUYzj1dwa.jpg'

# Display the image
Image(url=image_url, width = 575)

# **Context**

National Invitational Camp (NIC), more commonly known as the NFL Scouting Combine, began in 1982 when National Football Scouting, Inc. first conducted a camp for its member NFL clubs in Tampa, Florida. The key purpose then, same as it is today, was to ascertain medical information on the top draft eligible prospects in college football. The inaugural NIC was attended by a total of 163 players and established a foundation for future expansion.

As football and the art of evaluating players has evolved, so has the NFL Scouting Combine. While medical examinations remain the number one priority of the event, athletes will also participate in a variety of psychological and physical tests, as well as, formal and informal interviews with top executives, coaches and scouts from all 32 NFL teams. NIC is the ultimate four day job interview for the top college football players eligible for the upcoming NFL Draft.

Attribution: NFLCombine.net


# **About the Dataset**

This dataset contains 133 rows corresponding to a random sample of drafted players. A total of 9 variables are provided as listed below:

| Variable Name(s)      | Description                            |
|:----------------------|:--------------------------------------|
| Player                | Player ID, which is the player's name  |
| Pos                   | Position the player plays              |
| School                | College the player attended            |
| Ht                    | Player height (inches)                 |
| Wt                    | Player weight (lbs)                    |
| 40yd                  | Time to run the 40-yard dash (seconds) |
| Vertical              | Vertical jump height (inches)          |
| Broad Jump            | Horizontal distance covered (inches) (aka long jump) |
| Drafted (tm/rnd/yr)   | Team, round, and year the player was drafted |



Let's take a look at the data. To do this, first we import it directly from the url below.



# **A Snippet of the Data**

In [16]:
# Assigns the URL where the data file is stored to 'file_path'.
url='https://raw.githubusercontent.com/thamilton562/STAT108_Projects_Students/main/DataSets/NFLCombine.csv'

# Reads in the CSV data file and assigns it to the DataFrame 'df'.
df=pd.read_csv(url)

Next, we can display the data by *typing the name* of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [17]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)

# When you type the object name, the object gets printed.
df

Player   Pos       School  Ht   Wt  40yd  Vertical  Broad Jump  \
0      Myjai Sanders  EDGE   Cincinnati  77  228  4.67      33.0         120   
1    Keaontay Ingram    RB          USC  73  221  4.53      34.5         122   
2       Jesse Luketa    LB     Penn St.  75  253  4.89      37.5         114   
3      Marquis Hayes    OG     Oklahoma  77  318  5.30      23.5         102   
4      Troy Andersen    LB  Montana St.  76  243  4.42      36.0         128   
..               ...   ...          ...  ..  ...   ...       ...         ...   
128     Jahan Dotson    WR     Penn St.  71  178  4.43      36.0         121   
129   Brian Robinson    RB      Alabama  74  225  4.53      30.0         119   
130     Percy Butler     S    Louisiana  73  194  4.36      31.5         123   
131      Cole Turner    TE       Nevada  78  246  4.76      27.0         120   
132       Chris Paul    OG        Tulsa  76  323  4.89      27.0         109   

                                 Drafted (tm/rnd/yr)  
0        Arizona Cardinals / 3rd / 100th pick / 2022  
1        Arizona Cardinals / 6th / 201st pick / 2022  
2        Arizona Cardinals / 7th / 256th pick / 2022  
3        Arizona Cardinals / 7th / 257th pick / 2022  
4           Atlanta Falcons / 2nd / 58th pick / 2022  
..                                               ...  
128   Washington Commanders / 1st / 16th pick / 2022  
129   Washington Commanders / 3rd / 98th pick / 2022  
130  Washington Commanders / 4th / 113th pick / 2022  
131  Washington Commanders / 5th / 149th pick / 2022  
132  Washington Commanders / 7th / 230th pick / 2022  

[133 rows x 9 columns]

# **INSTRUCTIONS**

* Use Python to analyze the data set and complete each of the following.
* Replace ellipsis (...) with the relavent names or code.  
* For problems that require a written response, replace the ellipsis (...) by double clicking the text box to start typing.
* Reference the tutorial from activity for assistance.
* If you still need help:
 * Watch the video.
 * Attend office hours.

# **The variable to analyze**
You will analyze a category of a qualitative variable. Based on the first initial of your LAST name, analyze the category of the variable listed in the table. Use this category for the entire project.

| Last Name | Variable = Category |
|-----------|-------------------------------|
| A-L       | Position = Linebacker (LB)  |
| M-Z       | Position = Wide Receiver (WR)  |

In [18]:
# Print all the category names.
# Use this list to ensure correct spelling of your category.

# Use the following code for students.
#print("... category names")                #Replace ... with the variable name written out
#print("--------------------------------")
#freq_table = pd.Series(df['...']).value_counts()        #Replace ... with the variable name
#print(freq_table)

print("Position category counts")
print("--------------------------------")
freq_table = pd.Series(df['Pos']).value_counts()
print(freq_table)


Position category counts
--------------------------------
Pos
WR      20
RB      17
OG      17
LB      15
OT      15
EDGE    10
S        9
DE       6
DT       5
C        5
CB       5
QB       4
TE       4
P        1
Name: count, dtype: int64


# **QUESTION 1**
## Confidence Interval

**Last Names A-L:** Construct and interpret the 89% confidence interval for the population proportion of players at the combine who are linebackers.

**Last Names M-Z:** Construct and interpret the 89% confidence interval for the population proportion of players at the combine who are wide receivers.



**1.1) Parameter: Define the parameter, using correct notation.**

**A-L:** p is the population proportion of players at the combine who are linebackers.

**M-Z** p is the population proportion of players at the combine who are wide receivers.

**1.2) Method: Name the method you will use.**

1 sample z-interval for p

-OR-

1 proportion z-interval

**1.3) Assumptions:**

Complete the code below to find out how many majors fall under the category assigned to you.

In [19]:
# Count total observations
n = len(df)

#Use this code for students
# Count total successes
# Replace the 1st ... with the variable name
# Replace the 2nd ... with the name of the major category to be analyzed
#obs_count = df['...'].value_counts().get('...')

#print(f"{obs_count} out of {n} players at the combine are ...") #Replace ... with the position you are analyzing.

#---------------------------------
#A-L
obs_count_AL = df['Pos'].value_counts().get('LB')
print(f"{obs_count_AL} out of {n} players at the combine are linebackers.")

#---------------------------------
#M-Z
obs_count_MZ = df['Pos'].value_counts().get('WR')
print(f"{obs_count_MZ} out of {n} players at the combine are wide receivers.")

15 out of 133 players at the combine are linebackers.
20 out of 133 players at the combine are wide receivers.


**Show that both assumptions are met.**

**A-L:**

1. We will assume the sample represents the population (representative sample).

2. actual successes: 133(15/133) = 15 > 10

   actual failures: 133(1 - 15/100) = 118 > 10

   The sampling distribution of p̂ is approximately normal.

--------------------------------------------------------------
**M-Z:**

1. We will assume the sample represents the population (representative sample).

2. actual successes: 133(20/133) = 20 > 10

   actual failures: 133(1 - 20/133) = 113 > 10

   The sampling distribution of p̂ is approximately normal.

**1.4) Calculate: Complete the code below to calculate the sample proportion of majors that fall under the engineering category, and the confidence interval.**

In [20]:
# Define the confidence level
# Replace the ... with the stated confidence level, as a decimal (ex: 0.83, not 83%)
#CL = ...

#Use this code for students
#Calculate the values needed; p-hat, critical value (CV), and standard error (se).
#p_hat = obs_count / n
#cv = norm.ppf((1+CL)/2)
#se = np.sqrt(p_hat * (1-p_hat) / n)

#Calculate the bounds of the interval
#ci_lower = (p_hat - cv * se)
#ci_upper = (p_hat + cv * se)

#print(f"p-hat = {obs_count}/{n} = {p_hat.round(5)}")
#print(f"The {CL*100}% CI is ({ci_lower.round(5)}, {ci_upper.round(5)})")

#---------------------------------
#A-L
#Calculate the values needed; p-hat, critical value (CV), and standard error (se).
CL = 0.89
p_hat_AL = obs_count_AL / n
cv = norm.ppf((1+CL)/2)
se_AL = np.sqrt(p_hat_AL * (1-p_hat_AL) / n)

#Calculate the bounds of the interval
ci_lower_AL = (p_hat_AL - cv * se_AL)
ci_upper_AL = (p_hat_AL + cv * se_AL)     #ci_upper = (...) #Replace ... with the formula for the upper bound.

print("A-L:")
print(f"p-hat = {obs_count_AL}/{n} = {p_hat_AL.round(5)}")
print(f"The {CL*100}% CI is ({ci_lower_AL.round(5)}, {ci_upper_AL.round(5)})")

#---------------------------------
#M-Z
#Calculate the values needed; p-hat, critical value (CV), and standard error (se).
p_hat_MZ = obs_count_MZ / n
cv = norm.ppf((1+CL)/2)
se_MZ = np.sqrt(p_hat_MZ * (1-p_hat_MZ) / n)

#Calculate the bounds of the interval
ci_lower_MZ = (p_hat_MZ - cv * se_MZ)
ci_upper_MZ = (p_hat_MZ + cv * se_MZ)     #ci_upper = (...) #Replace ... with the formula for the upper bound.

print("\nM-Z:")
print(f"p-hat = {obs_count_MZ}/{n} = {p_hat_MZ.round(5)}")
print(f"The {CL*100}% CI is ({ci_lower_MZ.round(5)}, {ci_upper_MZ.round(5)})")

A-L:
p-hat = 15/133 = 0.11278
The 89.0% CI is (0.06895, 0.15662)

M-Z:
p-hat = 20/133 = 0.15038
The 89.0% CI is (0.10084, 0.19991)


**1.5) Communicate Results: Interpret the confidence interval calculated in 1.4 above. Round to three (3) decimal places.**

**A-L:** I am 89% confident that the interval 0.069 to 0.157 captures the population proportion of players at the combined who are linebackers.

**M-Z:** I am 89% confident that the interval 0.101 to 0.200 captures the population proportion of players at the combined who are wide receivers.

**1.6) Show work to calculate the margin of error. Then interpret the margin of error.**

**A-L:**
**Calculation:**

ME = (0.157 - 0.069)/2 = 0.044

**Interpretation:**

I am 89% confident that the population proportion of players at the combined who are linebackers differs from p̂ = 0.113 by at most 0.044.

-------------------------------------------------------
**M-Z:**
**Calculation:**

ME = (0.200 - 0.101)/2 = 0.099

**Interpretation:**

I am 89% confident that the population proportion of players at the combined who are wide receivers differs from p̂ = 0.150 by at most 0.099.



# **Question 2**

## **Hypothesis Test**

**A-L:** Is there convincing evidence that the population proportion of players at the combined who are linebackers is different from 10% (0.10)? Use α=0.11. Write up the solution using the PMACC procedure.

**M-Z:** Is there convincing evidence that the population proportion of players at the combined who are wide receivers is different from 25% (0.25)? Use α=0.11. Write up the solution using the PMACC procedure.

**2.1) Parameter: Define the parameter, using correct notation.**

**A-L:** p is the population proportion of players at the combine who are linebackers.

**M-Z** p is the population proportion of players at the combine who are wide receivers.

**2.2) Method: Name the method you will use, and write the hypotheses.**

**Method name:**

1 sample z-test for p

-OR-

1 proportion z-test

**Hypotheses:**

|**A-L**      |...| **M-Z**      |
|-------------|---|--------------|
| H0: p = 0.1 |...| H0: p = 0.25 |
| H1: p ≠ 0.1 |...| H1: p ≠ 0.25 |

**2.3) Assumptions: Show that both assumptions are met. Round to 1 decimal place.**

**A-L**

1. We will assume the sample represents the population (representative sample).

2. expected successes: 133(0.1) = 13.3 > 10

   expected failures: 133(1 - 0.1) = 119.7 > 10

   The sampling distribution of p̂ is approximately normal.

--------------------------------------------------------
**A-L**

1. We will assume the sample represents the population (representative sample).

2. expected successes: 133(0.25) = 33.25 > 10

   expected failures: 133(1 - 0.25) = 99.75 > 10

   The sampling distribution of p̂ is approximately normal.

**2.4) Calculate: Complete the code below to calculate the values required.**

In [21]:
#Use this code for students
#Define p0, the value in H0.
#p_0 = ... #Replace ... with p0.

#Calculate the values needed; p-hat, and standard error (se).
#p_hat = obs_count / n
#se = np.sqrt(p_0 * (1-p_0) / n)

#Calculate the z-score of our p-hat, under the assumption H0 is true.
#z_score = (p_hat - p_0) / se

#Calculate the p-value for 1- and 2-sided tests
#p_value1 = (1 - norm.cdf(abs(z_score)))
#p_value2 = 2 * p_value1

#print(f"p-hat = {obs_count}/{n} = {p_hat.round(7)}")
#print(f"z-score = {z_score.round(7)}")
#print(f"1 sided p-value = {p_value1:.11f}")
#print(f"2 sided p-value = {p_value2:.11f}")

#-------------------------------
#A-L
#Define P0, the value in H0.
#Replace ... with p0, the value in the null hypothesis.
p_0 = 0.1     #p_0 = ...

#Calculate the values needed; p-hat, and standard error (se).
p_hat_AL = obs_count_AL / n
se_AL = np.sqrt(p_0 * (1-p_0) / n)

#Calculate the z-score of our p-hat, under the assumption H0 is true.
z_score_AL = (p_hat_AL - p_0) / se_AL

#Calculate the p-value for 1- and 2-sided tests
p_value1_AL = (1 - norm.cdf(abs(z_score_AL)))
p_value2_AL = 2 * p_value1_AL

print("A-L:")
print(f"p-hat = {obs_count_AL}/{n} = {p_hat_AL.round(7)}")
print(f"z-score = {z_score_AL.round(7)}")
print(f"1 sided p-value = {p_value1_AL:.11f}")
print(f"2 sided p-value = {p_value2_AL:.11f}")

#-------------------------------
#M-Z
#Define P0, the value in H0.
#Replace ... with p0, the value in the null hypothesis.
p_0 = 0.25     #p_0 = ...

#Calculate the values needed; p-hat, and standard error (se).
p_hat_MZ = obs_count_MZ / n
se_MZ = np.sqrt(p_0 * (1-p_0) / n)

#Calculate the z-score of our p-hat, under the assumption H0 is true.
z_score_MZ = (p_hat_MZ - p_0) / se_MZ

#Calculate the p-value for 1- and 2-sided tests
p_value1_MZ = (1 - norm.cdf(abs(z_score_MZ)))
p_value2_MZ = 2 * p_value1_MZ

print("\nM-Z")
print(f"p-hat = {obs_count_MZ}/{n} = {p_hat_MZ.round(7)}")
print(f"z-score = {z_score_MZ.round(7)}")
print(f"1 sided p-value = {p_value1_MZ:.11f}")
print(f"2 sided p-value = {p_value2_MZ:.11f}")

A-L:
p-hat = 15/133 = 0.112782
z-score = 0.4913623
1 sided p-value = 0.31158510613
2 sided p-value = 0.62317021226

M-Z
p-hat = 20/133 = 0.1503759
z-score = -2.6533187
1 sided p-value = 0.00398522850
2 sided p-value = 0.00797045700


**2.5) Communicate Results: What conclusion is made about the null hypothesis? And what does that mean about the alternate hypothesis?**

**A-L:**
Because p-value = 0.623 > α = 0.11, we fail to reject H0. We do not have convincing evidence the population proportion of players at the combine who are linebackers is not equal to 0.10.

**M-Z:**
Because p-value = 0.008 < α = 0.11, we reject H0. We do have convincing evidence the population proportion of players at the combine who are wide receivers is not equal to 0.25.

# **Question 3**

## **Do you make the same conclusion if you use the confidence interval?**

**In question 2 you concluded that we either do have or do not have convincing evidence for the alternate hypothesis. Using your confidence interval from question 1, do you reach the same conclusion?**

**A-L:** Yes, 0.1 is in the 89% interval (0.06895, 0.15662), so we do not have convincing evidence that the population proportion of players at the combine who are linebackers is different from 0.1.

**M-Z** Yes, 0.25 is not in the 89% interval (0.10084, 0.19991), so we do have convincing evidence that the population proportion of players at the combine who are wide receivers is different from 0.25.

# **QUESTION 4**

Generate a paragraph of at least 100 words to address one of the following questions. That is, answer only 8a or 8b, but not both.

**8a)** Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

...

--OR--

**8b)** Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?

...


<br><br>
### Once you are done and ready to submit, follow the instructions below to save as a PDF and submit to GradeScope.

### Save as PDF
Note 1: You do not have to select Print Preview. You can print directly from the notebook.
Note 2: Image and graph sizes have been set so you should be able to see them correctly without making any changes to the browser width or the layout (portrait vs landscape).
1. Run all code one last time and make sure your graphs can be seen.
2. File -> Print (or ctrl-p/cmnd-p)
3. Change the "Desination" to PDF.
4. Save the PDF, taking note of where it is saved.

### Submit to GradeScope
**Watch the "GradeScope Submission" video for help.**
1. Login to the Canvas course
2. Click on GradeScope in the course navigation.
3. If you see multiple courses in GradeScope, click on the STAT 108 course
4. Click on the name of the assignment that matches your data set
5. Click on "Submit Work", select PDF
6. Select the PDF you just created
7. You need to tell GradeScope which page each problem answer/output is on. You should see a list of problems on the left, and a display of pages (thumbnails) on the right. Assign pages to questions by clicking on the question number on the left, then clicking on all pages that question is on.
8. After ALL questions have been assigned to their respective page(s), click "Submit"

#### **Still need help? Your STAT 108 team is here to help. Take your laptop to office hours.**
